### PR TITLE
Re-add environment variables field to template based lambda run configurations

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Re-add the field to template based lambda run configurations
- When selecting a template-based function, it will now auto-fill the environment variables the function has
- To make this work 100% correctly as SAM expects, we will need a new control with metadata of what the environment variable belongs to since we load and select asynchronously. I started work on this but we can revisit it in the future.

## Related Issue(s)
#2282

## Screenshots (if appropriate)

https://user-images.githubusercontent.com/11964782/103676240-b1d02000-4f35-11eb-87a3-1aedb3168889.mp4

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rli — APPROVED** [review]

**Comment by abrooksv** [comment]
Can we wrap the template and selector in another panel so it doesnt mess with the env var label?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt [hunk 1]
```diff
@@ -36,6 +37,25 @@ object SamTemplateUtils {
     private val MAPPER = ObjectMapper(YAMLFactory())
     private const val S3_URI_PREFIX = "s3://"
 
+    fun getFunctionEnvironmentVariables(template: Path, logicalId: String): Map<String, String> = readTemplate(template) {
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt:40]: Add unit tests for this please

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt [hunk 2]
```diff
@@ -36,6 +37,25 @@ object SamTemplateUtils {
...
+        val globalVars = try {
+            MAPPER.convertValue<Map<String, String>>(globals)
+        } catch (e: Exception) {
+            null
+        } ?: emptyMap()
+        val vars = try {
+            MAPPER.convertValue<Map<String, String>>(variables)
+        } catch (e: Exception) {
+            null
+        } ?: emptyMap()
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/sam/SamTemplateUtils.kt:54]: Can be simplified to `runCatching { MAPPER.convertValue<Map<String, String>>(variables) }.getOrDefault(emptyMap())`

**Review by hunterwerlla — COMMENTED** [review]
> Can we wrap the template and selector in another panel so it doesnt mess with the env var label?

Sure,
Screen Shot 2021-01-06 at 9 15 22 AM

**Review by abrooksv — APPROVED** [review]

**Pull request merged**